wagtail/wagtailcore/permission_policies/collections.py

```diff
@@ -0,0 +1,348 @@
+    applied to objects according to the standard django.contrib.auth permission model.
+    """
+    def user_has_permission(self, user, action):
+        """
+        Return whether the given user has permission to perform the given action
+        on some or all instances of this model
+        """
+        return self._check_perm(user, [action])
+
+    def user_has_any_permission(self, user, actions):
+        """
+        Return whether the given user has permission to perform any of the given actions
+        on some or all instances of this model
+        """
+        return self._check_perm(user, actions)
+
+    def users_with_any_permission(self, actions):
+        """
+        Return a queryset of users who have permission to perform any of the given actions
+        on some or all instances of this model
+        """
+        return self._users_with_perm(actions)
+
+    def user_has_permission_for_instance(self, user, action, instance):
+        """
+        Return whether the given user has permission to perform the given action on the
+        given model instance
+        """
+        return self._check_perm(user, [action], collection=instance.collection)
+
+    def user_has_any_permission_for_instance(self, user, actions, instance):
+        """
+        Return whether the given user has permission to perform any of the given actions
+        on the given model instance
+        """
+        return self._check_perm(user, actions, collection=instance.collection)
+
+    def instances_user_has_any_permission_for(self, user, actions):
+        """
+        Return a queryset of all instances of this model for which the given user has
+        permission to perform any of the given actions
+        """
+        if not user.is_active:
+            return self.model.objects.none()
+        elif user.is_superuser:
+            return self.model.objects.all()
+        else:
+            # filter to just the collections with this permission
+            return self.model.objects.filter(
+                collection__in=list(self._collections_with_perm(user, actions))
+            )
+
+    def users_with_any_permission_for_instance(self, actions, instance):
+        """
+        Return a queryset of all users who have permission to perform any of the given
+        actions on the given model instance
+        """
+        return self._users_with_perm(actions, collection=instance.collection)
+
+    def collections_user_has_any_permission_for(self, user, actions):
+        """
+        Return a queryset of all collections in which the given user has
+        permission to perform any of the given actions
+        """
+        if user.is_active and user.is_superuser:
+            # active superusers can perform any action (including unrecognised ones)
+            # in any collection
+            return Collection.objects.all()
+
+        elif not user.is_authenticated():
```

Comment:
Ah, good catch! Yep, I think that needs to be added - I guess I probably copied the logic from https://github.com/torchbox/wagtail/blob/master/wagtail/wagtailcore/permission_policies.py#L323 (which doesn't need an `is_authenticated` test because `has_perm` will return false for non-authenticated users).